hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -524,10 +525,15 @@ public static S3AFileStatus createFileStatus(Path keyPath,
       long blockSize,
       String owner,
       String eTag,
-      String versionId) {
+      String versionId,
+      boolean isCSEEnabled) {
```

Comment:
do you think it's time we moved to a builder here?

Reply:
Would require a few changes, don't know if it's worth it in this patch or should we do kind of a refactor patch separately?

Reply:
ok, for now keep separate

Reply:
if you want to do it....